### PR TITLE
add extension swap-errors

### PR DESCRIPTION
**Pull request by jgirardet**
the `swap-errors` extension allows swapping response when HTTP Error code is returned.
It's nice way to deal with errors server-side in
It works adding the header `HX-SWAP-ERRORS` in the response.

### CODE REVIEW

**Comment by 1cg** [comment]
@jgirardet do you think that this extension has been obviated by the new support for configuring swap behavior in the `htmx:beforeSwap` event?

**Comment by jgirardet** [comment]
No I don't. This extension allows you to write less javascript where htmx:beforeSwap forces to write more JavaScript.
Since many errors are handled server side, I think it's better for htmx to support the result (with a non 200 status code) out of the box.

File: src/ext/swap-errors.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+htmx.defineExtension("swap-errors", {
+  onEvent: function (name, evt) {
+    if (name === "htmx:beforeSwap") {
+      if (evt.detail.xhr.getResponseHeader("HX-SWAP-ERRORS") !== null) {
```
Comment (1cg) [inline src/ext/swap-errors.js:4]: Only change I would make is to not use all caps, which is more consistent w/ other htmx headers.

**Review by 1cg — CHANGES_REQUESTED** [review]

**Comment by alexpetros** [comment]
> This extension allows you to write less javascript where htmx:beforeSwap forces to write more JavaScript.
Since many errors are handled server side, I think it's better for htmx to support the result (with a non 200 status code) out of the box.

I agree with this, though coming up with a conceptual framework that allows configuration via attributes (instead of JS) is difficult!

We're working with this concept in #1620, so I'm going to close this PR in favor of that.

**Pull request closed without merging**